project/CodeStone1125/README.md

```diff
@@ -0,0 +1,166 @@
+their parents.
+
+Furthermore, quadtrees possess the unique ability to dynamically fine-tune 
+image compression precision, delicately balancing the level of detail with
+the consumption of computational resources. This adaptability is of paramount
+importance, especially in resource-constrained domains such as IoT and machine
+learning[1]. By elevating the threshold of the quadtree, we can craft images 
+with reduced intricacy in less critical areas, thus conserving valuable computing
+resources. Consequently, quadtrees emerge as invaluable assets in the realm of 
+image compression.
+
+### Why C++
+The rationale behind implementing this project in C++ is primarily driven by 
+the substantial computational demands.
+
+As the number of iterations increases, the growth in the number of leaf nodes
+within the Quadtree follows an exponential pattern. For each leaf node, we need
+to calculate `node.error` using the following formula:
+
+$$MSE = Σ[(Ri - R_avg)^2 + (Gi - G_avg)^2 + (Bi - B_avg)^2] / N$$
+
+Furthermore, within this [project](https://github.com/Inspiaaa/QuadTreeImageCompression#readme),
+it is evident that if one seeks an image compression
+that remains imperceptible to the human eye but reduces the file size significantly,
+it necessitates approximately 80,000 iterations, taking roughly 21 seconds for each
+image. I am confident that by implementing it in C++, I can enhance the acceleration
+of this process.
+| Original             | 80,000 Iterations          |
+| -------------------- | -------------------------- |
+| ![image](https://github.com/CodeStone1125/nsdhw_23au/assets/72511296/0ca3fa48-ad67-4084-87db-f5c820720f32)| ![image](https://github.com/Inspiaaa/QuadTreeImageCompression/blob/master/docs/sunset_80000.jpg) |
+
+## Prospective users
+
+* Machine Learning developer: image compressor is an important step for image preprocess,
+   It will accelerate the process of training model. And Quadtreebased image compressor
+    can decide the degree of compression dynamiclly. Especially suitable for the device
+  which need to strike ballence between speed and compution resources.
+
+* videographer: For 3D animation creators, it is necessary to reduce precision in less
+  critical areas, such as the background. This allows the animation to allocate computational
+  resources to more important elements. Additionally, the quadtree's recursive image
+  division characteristic imparts a distinct style, enabling artists to create unique
+   works using quadtree image compression.
+  
+| ![image](https://github.com/CodeStone1125/nsdhw_23au/assets/72511296/bf6fb00a-38ab-4a33-b07a-215f85d6484e)|
+|:-----------------------------------:|
+| **Quadtree art** |
+  
+
+## System Architecture
+
+|![SA2 drawio](https://github.com/CodeStone1125/nsdhw_23au/assets/72511296/b235b4d8-bfef-4f4d-9e9b-7225d1536116)|
+|:-----------------------------------:|
+| **System flow chart** |
+
+## API Description
+
+Python API:
```

Comment:
The API description is good enough as a plan.  This project can potentially use a much more tailored API design.  But you may do it as an extension.

A basic API design like what you have now is good for a term project.  But of course, do not prevent yourself from doing better 🙂

Reply:
I acknowledge the potential for a more customized API design in the future, particularly as an extension. If there are any plans for extensions, I will ensure their practical implementation in the project. Your input is greatly appreciated and will play a crucial role in my project enhancements.